packages/react-instantsearch-hooks-dom/src/ui/SearchBox.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+
+import { cx } from './lib/cx';
+
+import type {
+  ChangeEvent,
+  FormEvent,
+  RefObject,
+  JSXElementConstructor,
+} from 'react';
+
+export type SearchBoxProps = React.HTMLAttributes<HTMLDivElement> & {
+  inputRef: RefObject<HTMLInputElement>;
+  isSearchStalled: boolean;
+  onChange(event: ChangeEvent): void;
+  onReset(event: FormEvent): void;
+  onSubmit?(event: FormEvent): void;
+  placeholder?: string;
+  value: string;
+  resetIconComponent?: JSXElementConstructor<Record<string, any>>;
+  submitIconComponent?: JSXElementConstructor<Record<string, any>>;
+  loadingIconComponent?: JSXElementConstructor<Record<string, any>>;
```

Comment:
It would make more sense to accept elements instead of components here, don't you think?

Reply:
I went with components, as that way the classNames can be passed (in a later PR), but if we don't think that's important, can be changed to element as well I guess